src/Sarif.Multitool/ResultMatchSetCommand.cs

```diff
@@ -0,0 +1,129 @@
+
+using System;
+using System.Collections.Generic;
+using System.Diagnostics;
+using System.IO;
+using System.Linq;
+using Microsoft.CodeAnalysis.Sarif.Baseline.ResultMatching;
+using Newtonsoft.Json;
+
+namespace Microsoft.CodeAnalysis.Sarif.Multitool
+{
+    internal class ResultMatchSetCommand : CommandBase
+    {
+        private readonly IFileSystem _fileSystem;
+
+        public ResultMatchSetCommand(IFileSystem fileSystem = null)
+        {
+            _fileSystem = fileSystem ?? new FileSystem();
+        }
+
+        public int Run(ResultMatchSetOptions options)
+        {
+            options.OutputFolderPath = options.OutputFolderPath ?? Path.Combine(options.FolderPath, "Out");
+
+            ISarifLogMatcher matcher = ResultMatchingBaselinerFactory.GetDefaultResultMatchingBaseliner();
+            Formatting formatting = (options.PrettyPrint ? Formatting.Indented : Formatting.None);
+
+            // Remove previous results
+            if (Directory.Exists(options.OutputFolderPath))
+            {
+                Directory.Delete(options.OutputFolderPath, true);
+            }
+
+            // Create output folder
+            Directory.CreateDirectory(options.OutputFolderPath);
+
+            string previousFileName = "";
+            string previousGroup = "";
+            SarifLog previousLog = null, currentLog = null;
+
+            foreach (string filePath in Directory.GetFiles(options.FolderPath, "*.sarif"))
+            {
+                string fileName = Path.GetFileName(filePath);
+                string currentGroup = GetGroupName(fileName);
+
+                try
+                {
+                    currentLog = ReadSarifFile<SarifLog>(_fileSystem, filePath);
+
+                    // Compare each log with the previous one in the same group
+                    if (currentGroup.Equals(previousGroup) && currentLog?.Runs[0].Results.Count != 0 && previousLog?.Runs[0].Results.Count != 0)
+                    {
+                        Console.WriteLine();
+                        Console.WriteLine($"{previousFileName} -> {fileName}:");
+                        SarifLog result = matcher.Match(new[] { previousLog }, new[] { currentLog }).First();
+
+                        // Write the same and different count and different IDs
+                        WriteDifferences(result);
+
+                        // Write the log, if there were any changed results
+                        if (result.Runs[0].Results.Any(r => r.BaselineState != BaselineState.Unchanged))
+                        {
+                            string outputFilePath = Path.Combine(options.OutputFolderPath, fileName);
+                            WriteSarifFile(_fileSystem, result, outputFilePath, formatting);
+                        }
+                    }
+                }
+                catch (Exception ex) when (!Debugger.IsAttached)
+                {
+                    Console.WriteLine(ex.ToString());
+                }
```

Comment:
>} [](start = 16, length = 1)

Suppose we have MyGroup-001.sarif, MyGroup-002.sarif, MyGroup-003.sarif. We read 001 successfully, but we throw an exception while attempting to process 002 and compare it to 001. Now, says the code, we keep going and attempt to compare 003 to 002. Is that the right thing to do?

Alternatively, we could:
- Give up entirely.
- Give up on 002, but now try to recover by comparing 003 to 001.

In the latter case, we would put lines 75-77 below _inside_ the `try` block, so we only advance the "previous log" if we processed it successfully.

Thoughts?